### PR TITLE
Feature: IPv6 Support

### PR DESCRIPTION
**Pull request by poelzi**
**What this PR does / why we need it**:

Support for IPv6 which is widely used and should be supported.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Todo:
- infoblox support, requires next version of infoblox go client version which is not yet released.

**Release note**:

```feature operator
Basic IPv6 Support
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @poelzi for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/controller/provider/infoblox/access.go [hunk 1]
```diff
@@ -71,6 +72,8 @@ func (this *access) NewRecord(fqdn string, rtype string, value string, zone prov
 			//Zone:     zone.Key(),
 			View: this.view,
 		}))
+	case dns.RS_AAAA:
+		log.Printf("warning: aaaa records not supported on infoblox yet")
```
Comment (MartinWeindel) [inline pkg/controller/provider/infoblox/access.go:76]: It is not sufficient to print a log line. There should be an explicit error and it should be reported in the DNSEntry status.

File: pkg/dns/provider/entry.go [hunk 2]
```diff
@@ -675,22 +686,24 @@ func normalizeTargets(logger logger.LogContext, object *dnsutils.DNSEntryObject,
...
+		if ip.To4() != nil {
+			ipv4addrs = append(ipv4addrs, ip.String())
+		} else if ip.To16() != nil {
+			ipv6addrs = append(ipv6addrs, ip.String())
 		}
-		addrs = append(addrs, ip.String())
 	}
-	if len(addrs) == 0 {
-		return nil, fmt.Errorf("%s has no IPv4 address (of %d addresses)", hostname, len(ips))
+	if len(ipv4addrs) == 0 || len(ipv6addrs) == 0 {
```
Comment (MartinWeindel) [inline pkg/dns/provider/entry.go:703]:
```suggestion
	if len(ipv4addrs) == 0 && len(ipv6addrs) == 0 {
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
Some minor changes are needed. I will also run some more tests in the next days

File: pkg/controller/provider/infoblox/access.go (hunk 1 above)
Comment (poelzi) [reply pkg/controller/provider/infoblox/access.go]: I changed the interface to include errors and propagated it into the status.

File: pkg/dns/provider/entry.go (hunk 2 above)
Comment (poelzi) [reply pkg/dns/provider/entry.go]: good catch. fixed

**Review by MartinWeindel — APPROVED** [review]
Functional tests need some adjustments and infoblox should be updated to new v2.
But this will be addressed in a separate PR.

Thanks for your contribution!

**Pull request merged**